### ISSUE TITLE
I have a custom post type selected as the source but it is pulling in another random post type as well

### ISSUE THREAD

**Issue by joelnewcomer** [dudaster/ele-custom-skin#82]
Hi! I am using the Posts widget with Elementor Custom Skin Pro to display a custom post type ('products'). I recently installed the Style Kits for Elementor plugin (https://wordpress.org/plugins/analogwp-templates/) and for some reason, the Posts widget is showing the style kit post type ('ang_tokens') in the loop. To track down the source of the issue, I began by deactivating Elementor Custom Skin and the issue went away. The website we are building is at this URL: http://blue1energy2.wpengine.com/ and you will see the issue if you scroll down to the "Featured Products" section. I even tried to exclude that post and it wouldn't exclude it. Please check into this issue for me ASAP. Thanks! Other than this, I've been loving Elementor Custom Skin Pro! Thanks!

Joel

**Comment by dudaster** [comment]
What does it say in query tab?

**Comment by joelnewcomer** [comment]
It says Source: Products.  See attached screenshot.
![query](https://user-images.githubusercontent.com/171541/66410274-d7cbe500-e9bf-11e9-98b4-bd99304bbbf7.png)

**Comment by dudaster** [comment]
And if you use classic skin this would not be an issue is it?

**Comment by joelnewcomer** [comment]
Correct. There is no issue when I switch to classic skin.

**Comment by dudaster** [comment]
It is weird indeed. Never used this plugin before. I need to see what it does and why is messing up with the query. It is not something like Envato Elements where you get what you want and then you can deactivate it, is it?

**Comment by dudaster** [comment]
Can you try exclude current post?

**Comment by joelnewcomer** [comment]
No, unfortunately it's not something I can use and then deactivate. I tried excluding the current post and the issue persists.

**Comment by dudaster** [comment]
I'll need to investigate further. I'll let you know when I'll find the issue and come with a fix.

**Comment by joelnewcomer** [comment]
Thanks man! Much appreciated!

**Comment by dudaster** [comment]
It is so wierd:
```
    echo get_the_ID();// prints the id of the current item
		$return = \Elementor\Plugin::instance()->frontend->get_builder_content_for_display( $template_id );// elementor function that generates an elementor template
    echo get_the_ID();// prints a random id
```

**Comment by joelnewcomer** [comment]
ahhh... so does that mean that this is an Elementor bug?

**Comment by dudaster** [comment]
I don't think so... if I delete all the style kits it's working.

**Comment by joelnewcomer** [comment]
Very odd ... so I guess that means that Style Kits for Elementor is to blame and I need to contact their tech support?

**Comment by dudaster** [comment]
Must be an hook from analogwp somewhere that interferes with the wp_query and let's it hanging.

**Comment by joelnewcomer** [comment]
I really appreciate you looking into this so quickly! You're the man!

**Comment by dudaster** [comment]
I'll try to contact them. And give you a link to the issue. Hopefully they will investigate the issue too. Meanwhile I'll keep on looking for what could trigger such issue.

It happens only for the first item and then it works fine, that's why you tried to exclude and still appears.

**Comment by dudaster** [comment]
https://wordpress.org/support/?post_type=topic&p=12009978 I wrote an invitation on their support forums... it will take a while to appear since I'm moderated (long story)

**Comment by JohnPixle** [comment]
Hey friends! John from Style Kits here. Thanks for bringing this up to our attention. I will notify Ram, the developer and will get back to you. Just wanted to let you guys know that we got the heads-up and will look into it asap.

Will check-in again soon. ✌🏻

**Comment by joelnewcomer** [comment]
Great! Thanks John!

**Comment by 4xd-co** [comment]
Hi! The same issue appeared on our live website here: https://www.minicards.hu/en/testimonials/
(see "Global Styles" on the top). The page was ok, then this happened unexpectedly. I tried to roll-back AnalogWP Style Kits to a previous version, but that didn't help, the issue persisted on the older versions, too.

**Comment by 4xd-co** [comment]
I managed to fix the issue by rolling back Ele Custom Skin to version 1.3.3 on our live site. You can see the issue with the latest version of Ele Custom Skin (v1.3.4) here: https://mc-demo.4xd.work/en/testimonials/

Does it mean that this is caused by Ele Custom Skin, and the bug is in the latest version?

**Comment by joelnewcomer** [comment]
@4xd-co awesome! Thanks for figuring that out.

**Comment by 4xd-co** [comment]
Now that v1.3.6 came out, this bug is still there. Had to roll back to v1.3.3 again. @dudaster when do you think it will be fixed?

**Comment by dudaster** [comment]
I'll work to exclude their post type hopefully will solve this problem. But sincerly the ball is in @JohnPixle field.

**Comment by mauryaratan** [comment]
@4xd-co This is Ram, from AnalogWP. I'm seeking some details regarding the issue to see if this is an issue at our end. I'd highly appreciate if you can get in touch via https://analogwp.com/contact/ or let me know what's the best way to reach you. Thanks.

**Comment by 4xd-co** [comment]
@mauryaratan Thanks, I sent a message via https://analogwp.com/contact/.

**Comment by dudaster** [comment]
@mauryaratan same issue happens without ECS.
Create a single elementor template and try and preview as a page.

**Comment by mauryaratan** [comment]
Thanks for replicating @dudaster, by any chance did you have ECS deactivated? Just trying to be double sure here if we have a conflict in middle :)
We've built all pages of https://analogwp.com on top of our own plugin, [single post template](https://analogwp.com/customize-elementor-column-gaps/) for instance without any issues so far. So possibly it is just a conflict.

The issue is really weird. I went ahead and made sure all `WP_Query` calls are followed by `wp_reset_postdata()`, other than that we're not modifying the query at all at any point. Speaking of that I noticed one of your `WP_Query` calls is [missing](https://plugins.trac.wordpress.org/browser/ele-custom-skin/trunk/skins/skin-custom.php#L140) that, not that it's the problem, just wanted to let you know.

Moreover, the post type that is leaked in preview of ECS, `ang_tokens` isn't actually publicly queryable if you were to look at `get_post_types( [ 'public' => true ] )` it won't return our CPT.

I'm still looking into the issue and will post here when I find a solution. Thanks.

**Comment by mauryaratan** [comment]
@dudaster I got it [fixed](https://github.com/mauryaratan/analogwp-templates/commit/972e1d02e3f35888707060be8b8d65a1760eda9d). Looks like the problem was with a `WP_Query` call inside the editor. Although there is nothing really wrong with the call there it somehow manages to screw things up.

I simply replaced it with `get_posts` function and things are working fine. An update is going live today/tomorrow among other changes fixing this issue.

cc @joelnewcomer @4xd-co
